src/main/java/io/vinyldns/java/model/batch/ChangeInput.java

```diff
@@ -39,18 +50,21 @@ public RecordType getType() {
     return type;
   }
 
+  public RecordData getRecord() { return record; }
+
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     ChangeInput that = (ChangeInput) o;
     return this.inputName.equals(that.inputName)
         && this.changeType == that.changeType
-        && this.type == that.type;
+        && this.type == that.type
+        && this.record == that.record;
```

Comment:
`==` in java for objects tests that they point to the same reference, you will want `this.record.equals(that.record)` here.

Also, keep in mind `this.record` and `that.record` could be null, so simply `this.record.equals(that.record)` is not safe.  Need a stronger check...

```java
boolean recordsAreEqual = ((this.record != null && this.record.equals(that.record)) || (that.record != null && that.record.equals(this.record)) || (this.record == null && that.record == null))
```

Reply:
Make sure once you update that you add some good unit tests around that

Reply:
Hmmm, down below we see `Objects.equals`, maybe that is the right approach?